### PR TITLE
Add simple usage example to README.

### PR DESCRIPTION
**Pull request by jcsackett**

### CODE REVIEW

**Comment by jcsackett** [comment]
QA:

Verify that the example provided actually works for you.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/6/
Test PASSed.

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,7 @@ help:
 	@echo "dev - install theblues in develop mode"
 	@echo "lint - check style with flake8"
 	@echo "test - run tests quickly with the default Python"
-	@echo "check - run tests against 2.7 and 3.4 and check lint."
+	@echo "check - clean env, run tests against 2.7 and 3.4, check lint,"
 	@echo "test-all - run tests against 2.7 and 3.4, check lint, and test docs."
 	@echo "coverage - check code coverage quickly with the default Python"
```
Comment (fabricematrat) [inline Makefile:16]: I think it's test-coverage or you should rename the target in the Makefile.

File: README.rst [hunk 2]
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
...
 Without these, theblues cannot communicate with the charmstore.
+
+Usage
+-----
+
+Interacting with the charmstore is pretty simple. To look up an entity on the
+charmstore (e.g. a charm or bundle)::
+
+    >>> from theblues.charmstore import CharmStore
+    >>> cs = CharmStore('http://api.jujucharms.com/v4')
```
Comment (fabricematrat) [inline README.rst]: I would put https here.

File: README.rst [hunk 3]
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
...
+    u'https://api.jujucharms.com/v4/wordpress/archive/hooks/install
+    >>> hook = cs.files('wordpress', filename='hooks/install', read_file=True)
+    >>> print hook
+    #!/bin/bash
+
+    set -xe
+    ...
+    <snipped for length>
+    ...
+    juju-log "So, environment is setup. We'll wait for some hooks to fire off before we get all crazy"
```
Comment (fabricematrat) [inline README.rst:52]: :)

**Comment by fabricematrat** [comment]
:+1:  with minor comments.
For future ref, I think we need to think not only about methods docs but also the data format so people knows where to find things.

File: Makefile [hunk 4]
```diff
@@ -11,7 +11,7 @@ help:
 	@echo "dev - install theblues in develop mode"
 	@echo "lint - check style with flake8"
 	@echo "test - run tests quickly with the default Python"
-	@echo "check - run tests against 2.7 and 3.4 and check lint."
+	@echo "check - clean env, run tests against 2.7 and 3.4, check lint,"
```
Comment (bac) [inline Makefile:14]: why the final comma? none of the others have trailing punctuation.

File: README.rst [hunk 5]
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
 	$apt-get install libmacaroons0 python-macaroons libsodium13
```
Comment (bac) [inline README.rst]: These packages come from a PPA.  I think it should be mentioned.

**Comment by bac** [comment]
:+1: and they all work!  QA-OK

File: Makefile (hunk 1 above)
Comment (jcsackett) [inline Makefile:16]: That is indeed a mismatch--thanks for spotting it. I've renamed the target to "coverage", to match targets like "check", "lint", etc.

File: README.rst (hunk 2 above)
Comment (jcsackett) [inline README.rst]: Sure thing. Done.

File: README.rst (hunk 3 above)
Comment (jcsackett) [inline README.rst:52]: The best part is that's the actual text from the hook. :P I kind of love that they wrote that in their install hook.

**Comment by jcsackett** [comment]
@fabricematrat thanks for the review! I've addressed your comments, I think.

> For future ref, I think we need to think not only about methods docs but also the data format so people knows where to find things.

I concur. I'm hoping we have public charmstore docs at some point covering that we can point to from theblues docs, rather than reproducing all of that information, but either way we need it.

This also raises the point that our library right now is little more than a wrap around a standard GET on the charmstore--at some point we may want to consider a higher level module that provides a nicer, more intuitive object.

File: Makefile (hunk 4 above)
Comment (jcsackett) [inline Makefile:14]: That's weird--the actual file reads "clean env, run tests against 2.7 and 3.4, check lint and test docs." It does have a trailing period, which I've removed. Not sure why the diff is clipping that last part.

File: README.rst (hunk 5 above)
Comment (jcsackett) [inline README.rst]: I did not know that. Added, and thanks for catching that.

**Comment by jcsackett** [comment]
@bac Thanks for the review and the QA.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/7/
Test PASSed.

**Comment by jcsackett** [comment]
Thanks again @fabricematrat and @bac. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/8/
Test PASSed.

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/theblues-lib-merge/8

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Pull request merged**